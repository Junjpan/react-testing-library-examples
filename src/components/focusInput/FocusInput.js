import React from 'react';

const FocusInput=()=>{

    const inputRef=React.useRef(null);
    return(
        <div>
            <input type="text" arial-label='focus-input' ref={inputRef} placeholder="Focus"/>
            <button onClick={()=>inputRef.current.focus()} >Click to focus </button>
        </div>
    )
}

export default FocusInput