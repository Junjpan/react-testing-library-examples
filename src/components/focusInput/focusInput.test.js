import React from 'react';
import FocusInput from './FocusInput';
import {render,fireEvent,cleanup} from '@testing-library/react';

test('FocusInput match snapshop',()=>{
    const {container}=render(<FocusInput />);
    expect(container.firstChild).toMatchSnapshot()
})

test('Clicking on button trigger focus on input',()=>{
    const {getByPlaceholderText,getByText}=render(<FocusInput />);
    const button=getByText('Click to focus');
    const focusInput=getByPlaceholderText('Focus');
    fireEvent.click(button);
    expect(document.activeElement).toBe(focusInput);
})