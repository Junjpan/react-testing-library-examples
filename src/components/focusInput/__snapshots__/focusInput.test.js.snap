// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusInput match snapshop 1`] = `
<div>
  <input
    arial-label="focus-input"
    placeholder="Focus"
    type="text"
  />
  <button>
    Click to focus 
  </button>
</div>
`;
